// nativefier/lib/options/fields/userAgent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.userAgent = void 0;
const log = require("loglevel");
const inferChromeVersion_1 = require("../../infer/browsers/inferChromeVersion");
const inferFirefoxVersion_1 = require("../../infer/browsers/inferFirefoxVersion");
const inferSafariVersion_1 = require("../../infer/browsers/inferSafariVersion");
const optionsMain_1 = require("../optionsMain");
const USER_AGENT_PLATFORM_MAPS = {
    darwin: 'Macintosh; Intel Mac OS X 10_15_7',
    linux: 'X11; Linux x86_64',
    win32: 'Windows NT 10.0; Win64; x64',
};
const USER_AGENT_SHORT_CODE_MAPS = {
    edge: edgeUserAgent,
    firefox: firefoxUserAgent,
    safari: safariUserAgent,
};
async function userAgent(options) {
    if (!options.nativefier.userAgent) {
        // No user agent got passed. Let's handle it with the app.userAgentFallback
        return null;
    }
    if (!Object.keys(USER_AGENT_SHORT_CODE_MAPS).includes(options.nativefier.userAgent.toLowerCase())) {
        // Real user agent got passed. No need to translate it.
        log.debug(`${options.nativefier.userAgent.toLowerCase()} not found in`, Object.keys(USER_AGENT_SHORT_CODE_MAPS));
        return null;
    }
    options.packager.platform = optionsMain_1.normalizePlatform(options.packager.platform);
    const userAgentPlatform = USER_AGENT_PLATFORM_MAPS[options.packager.platform === 'mas' ? 'darwin' : options.packager.platform];
    const mapFunction = USER_AGENT_SHORT_CODE_MAPS[options.nativefier.userAgent];
    return await mapFunction(userAgentPlatform, options.packager.electronVersion);
}
exports.userAgent = userAgent;
async function edgeUserAgent(platform, electronVersion) {
    const chromeVersion = await inferChromeVersion_1.getChromeVersionForElectronVersion(electronVersion);
    return `Mozilla/5.0 (${platform}) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${chromeVersion} Safari/537.36 Edg/${chromeVersion}`;
}
async function firefoxUserAgent(platform) {
    const firefoxVersion = await inferFirefoxVersion_1.getLatestFirefoxVersion();
    return `Mozilla/5.0 (${platform}; rv:${firefoxVersion}) Gecko/20100101 Firefox/${firefoxVersion}`;
}
async function safariUserAgent(platform) {
    const safariVersion = await inferSafariVersion_1.getLatestSafariVersion();
    return `Mozilla/5.0 (${platform}) AppleWebKit/${safariVersion.webkitVersion} (KHTML, like Gecko) Version/${safariVersion.version} Safari/${safariVersion.webkitVersion}`;
}
