// nativefier/lib/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PLACEHOLDER_APP_DIR = exports.ELECTRON_MAJOR_VERSION = exports.DEFAULT_SAFARI_VERSION = exports.DEFAULT_FIREFOX_VERSION = exports.DEFAULT_CHROME_VERSION = exports.DEFAULT_ELECTRON_VERSION = exports.DEFAULT_APP_NAME = void 0;
const path = require("path");
exports.DEFAULT_APP_NAME = 'APP';
// Update both DEFAULT_ELECTRON_VERSION and DEFAULT_CHROME_VERSION together,
// and update app / package.json / devDeps / electron to value of DEFAULT_ELECTRON_VERSION
exports.DEFAULT_ELECTRON_VERSION = '12.0.10';
exports.DEFAULT_CHROME_VERSION = '89.0.4389.128';
// Update each of these periodically
// https://product-details.mozilla.org/1.0/firefox_versions.json
exports.DEFAULT_FIREFOX_VERSION = '89.0';
// https://en.wikipedia.org/wiki/Safari_version_history
exports.DEFAULT_SAFARI_VERSION = {
    majorVersion: 14,
    version: '14.0.3',
    webkitVersion: '610.4.3.1.7',
};
exports.ELECTRON_MAJOR_VERSION = parseInt(exports.DEFAULT_ELECTRON_VERSION.split('.')[0], 10);
exports.PLACEHOLDER_APP_DIR = path.join(__dirname, './../', 'app');
